src/main/java/org/folio/service/processing/reader/SourceReaderBuilder.java

```diff
@@ -31,14 +34,14 @@ public static SourceReader build(File file, JobProfileInfo jobProfile) {
 
     if (isMarc(jobProfile)) {
       if (JSON_EXTENSION.equals(extension)) {
```

Comment:
@RuslanLavrov  I think it will be better to use switch-case here for more readability and code efficiency. At least, for now. What do you think?

Reply:
merging for now as is to be in time for hosted envs build